compiler/include/AggregateType.h

```diff
@@ -22,6 +22,7 @@
 
 #include "DecoratedClasses.h"
 #include "type.h"
+#include "../resolution/callInfo.h"
```

Comment:
AggregateType.cpp uses the 'call' field and 'toString' method from 'CallInfo', so a forward declaration won't work because it doesn't provide enough information.

I could change those methods to take two arguments, a CallExpr and char*, if you prefer.

Reply:
Normally the forward declaration is only used in the header. I.e. AggregateType.h has `class CallInfo` and AggregateType.cpp has the `#include "../resolution/callInfo.h".

Nonetheless if all you are getting from CallInfo is the name of the called function, I'd prefer that these methods accept that separately, because I view CallInfo as intended to be internal to resolution.

Reply:
OK, I'll look into passing those as separate args.